code/game/objects/items/devices/traitordevices.dm

```diff
@@ -382,13 +386,35 @@ effective or pretty fucking useless.
 	interacting_with.balloon_alert(user, "radio distrupted!")
 	to_chat(user, span_notice("You release a directed distruptor wave, disabling all radio devices on [interacting_with]."))
 	disable_radios_on(interacting_with)
+	disable_suit_sensors(interacting_with)
+	disable_gps_trackers(interacting_with)
+	disable_beacon_trackers(interacting_with)
 
 	return ITEM_INTERACT_SUCCESS
 
 /obj/item/jammer/proc/disable_radios_on(atom/target)
-	for (var/obj/item/radio/radio in target.get_all_contents() + target)
+	for(var/obj/item/radio/radio in target.get_all_contents() + target)
 		radio.set_broadcasting(FALSE)
 
+/obj/item/jammer/proc/disable_gps_trackers(atom/target)
+	for(var/obj/item/gps/gps in target.get_all_contents() + target)
+		var/datum/component/gps/item/gps_component = gps.GetComponent(/datum/component/gps/item)
+		if(gps_component.tracking)
+			gps.cut_overlay("working")
+			gps_component.tracking = FALSE
+
+/obj/item/jammer/proc/disable_beacon_trackers(atom/target)
+	for(var/obj/item/beacon/beacon in target.get_all_contents() + target)
+		beacon.turn_off()
+
+/obj/item/jammer/proc/disable_suit_sensors(atom/target)
+	for(var/obj/item/clothing/under/uniform in target.get_all_contents() + target)
+		if((uniform.has_sensor == NO_SENSORS) || !uniform.sensor_mode)
```

Comment:
Calling `get_all_contents` 4 times seems a little egregious when we could just call it once

Reply:
Refactored with https://github.com/tgstation/tgstation/pull/87931/commits/b53248df96221261f392664ac105c8706d000d16